### PR TITLE
add gbs_plex option to genotype_call_results_reporter

### PR DESCRIPTION
**Pull request by ces**
It's likely PSD will only utilise results from human QC plexes within SM and therefore adding an option so only those results can be posted back to their API.

### CODE REVIEW

File: t/70-report-genotype_call.t [hunk 1]
```diff
@@ -7,7 +7,7 @@ use Moose::Meta::Class;
 use npg_testing::db;
 use REST::Client;
 use JSON;
-
+use Data::Dump qw[pp];
```
Comment (dkj) [inline t/70-report-genotype_call.t:10]:
```suggestion
```
superfluous, right?

**Review by dkj — COMMENTED** [review]
Is there a test to check the filtering out? (like `never_used_plex` returning no results?)

File: t/70-report-genotype_call.t (hunk 1 above)
Comment (ces) [reply t/70-report-genotype_call.t]: Arrgh! thanks - resolved.

Comment (dkj) [reply t/70-report-genotype_call.t]: And yes there is such a test...

**Comment by ces** [comment]
> Is there a test to check the filtering out? (like `never_used_plex` returning no results?)

Yes but the test message was poor - now updated.

**Review by dkj — APPROVED** [review]
We could generalise this to take mulitplex plex (type `ArrayRef[Str]`?) if needed.

**Pull request merged**